packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -320,15 +320,15 @@ function commitHookEffectListUnmount(tag: number, finishedWork: Fiber) {
         const destroy = effect.destroy;
         effect.destroy = undefined;
         if (destroy !== undefined) {
-          destroy();
```

Comment:
Hm. This seems like it was originally an oversight. Unfortunate that we have to slow down even more. I'm not sure why we have this since we don't have the equivalent for mount effects. Want to open a separate PR for this change so we can discuss it there?

It doesn't actually save you from anything here since if `componentDidMount` throws, it'll abort the sequence anyway and possibly leave things uncalled.